### ISSUE TITLE
add isEqualToIgnoringGivenFieldsRecursive

### ISSUE THREAD

**Issue by vguna** [joel-costigliola/assertj-core#698]
Hi.

I always stumble across the missing option to check for equality ignoring given fields, but do it recursively. Could something like that be added easily?

For specifying fields to be ignore, one could just use the object graph notation for specifying ignores on deeper levels such as:

``` java
isEqualToIgnoringGivenFieldsRecursive(person, "name", "address.steetNo");
```

wdyt?

**Comment by joel-costigliola** [comment]
Yes, that would be useful I think, I'm thinking to support this:
```java
assertThat(frodo).isEqualToComparingFieldByFieldRecursivelyIgnoringGivenFields(sam, "age");
```
or a more general
```java
assertThat(frodo).usingRecursiveFieldByFieldComparator()
                 .isEqualToIgnoringGivenFields(sam, "age");
```

**Comment by vguna** [comment]
+1 for option 2 :).

**Comment by welandaz** [comment]
Hi @joel-costigliola, @vguna!

Is someone already working on this feature? If not, I can try to drop a pull request to support this kind of mechanism (not sure what your contribution policy is and whether you have only selected number of contributors)

Thanks!

**Comment by joel-costigliola** [comment]
Hey @vguna, no one is working on this feature AFAIK, you are welcome to drop a PR, when you do so you will have some guidance but that basically boils down to:

- have some javadoc with code examples to describe the feature so that it is easy for users to find out what it does (only for public facing API, no need for internal classes).
- write some tests (obvious right), using Assert assertions is very welcome :smile_cat:, create a new file for testing the new feature (see the existing  tests as guidance)
- try to respect the formatting rules (see https://github.com/joel-costigliola/assertj-core/tree/master/src/ide-support)
- if you have time, contribute an example of the new feature in assertj-examples on the [with-latest-snapshot-versions](https://github.com/joel-costigliola/assertj-examples/tree/with-latest-snapshot-versions) branch.
 
We will give you feedback but sometimes with a bit of delay (we are pretty busy these days !), don't hesitate to ask again for feedback if you see we haven't reviewed the PR.

ThaNks mate !

**Comment by welandaz** [comment]
Cool!

Just one thing to clarify: this should go to the branch 2.x  (Java 7 compatibility), right? ( not master )

**Comment by joel-costigliola** [comment]
correct :)

**Comment by joel-costigliola** [comment]
closed as this will be done as part of #1002.
